compiler/util/misc.cpp

```diff
@@ -98,6 +98,15 @@ void setupDynoError(chpl::ErrorBase::Kind errKind) {
   exit_eventually |= err_fatal;
 }
 
+GpuCodegenType getGpuCodegenType() {
+  INT_ASSERT(usingGpuLocaleModel());
+  if (0 == strcmp(CHPL_GPU_CODEGEN, "cuda")) {
+    return GPU_CG_CUDA;
+  } else {
+    return GPU_CG_AMD;
```

Comment:
I guess in practice `printchplenv` will already ensure that `CHPL_GPU_CODEGEN` is "cuda" or "amd".

My kneejerk defensive programming reaction can't help but say you  should add:

```
INT(!strcmp(CHPL_GPU_CODEGEN, "amd"))
```

to your else branch though.

I guess this is arguably premature optimization, but it strikes me that that this function is called multiple times, will always compute the same result (CHPL_GPU_CODEGEN can't change during compilation), and does string comparisons. The is the sort of thing that my kneejerk optimization reaction says we should cache.

I suspect something like this would work:

```C++
GpuCodegenType getGpuCodegenType() {
  static const GpuCodegenType cached = []() {
    if(!strcmp(CHPL_GPU_CODEGEN, "cuda")) { return CUDA; }
    INT_ASSERT(!strcmp(CHPL_GPU_CODEGEN", "amd");
    return GpuCodegenType::AMD;
  }();
  return cached;
}
```

If you think that obfuscates things (or won't work the way I think it does) feel free to not take it; I wouldn't think this function would be called in a tight loop.

Reply:
Never seen a static lambda call for deferring computation. I like it. Thanks!